lib/Connector/OpenWeatherMapConnector.php

```diff
@@ -638,4 +642,52 @@ private static function cardinalDirections()
             'NW' => [292.5, 337.5]
         ];
     }
+
+    /**
+     * @param ScheduleCriteriaRequestInterface $event
+     * @return void
+     * @throws ConfigurationException
+     */
+    public function onScheduleCriteriaRequest(ScheduleCriteriaRequestInterface $event): void
+    {
+        // Initialize Open Weather Schedule Criteria parameters
+        $event->addType('open_weather_map', __('Open Weather Map'))
```

Comment:
I can see the reasoning for having `open_weather_map` as the type, however this would make it harder to replace Open Weather Map with a different service.

If we made the type `weather`, then another service could also implement these criteria via another Connector, allowing Open Weather Map to be replaced.